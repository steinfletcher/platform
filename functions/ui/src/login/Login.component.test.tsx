import {shallow} from 'enzyme';

import {LoginForm} from "./Login.component";
import * as React from "react";
import {noop} from "../shared/lang";

const noopActions = {
    updateUsername: noop,
    updatePassword: noop,
    onCreateSession: noop,
};

describe('LoginForm', () => {
    it('should render', () => {
        const wrapper = shallow(
            <LoginForm
                disabled={false}
                username={"a@b.com"}
                password={"Password123"}
                loginPending={false}
                {...noopActions}
            />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should disable submit when login pending', () => {
        const wrapper = shallow(
            <LoginForm
                disabled={true}
                username={"a@b.com"}
                password={"Password123"}
                loginPending={true}
                {...noopActions}
            />);

        const submitButton = wrapper.find('[data-test-id="submitBtn"]');

        expect(submitButton.prop("disabled")).toBeTruthy();
        expect(submitButton.text()).toContain('Loading...');
    });

    it('should display an error when error message is present', () => {
        const wrapper = shallow(
            <LoginForm
                disabled={false}
                username={"a@b.com"}
                password={"Password123"}
                loginPending={false}
                errorMessage={"wrong password"}
                {...noopActions}
            />);

        const errorMessage = wrapper.find('[data-test-id="incorrectCredentials"]');

        expect(errorMessage.text()).toEqual("wrong password");
    });
});
