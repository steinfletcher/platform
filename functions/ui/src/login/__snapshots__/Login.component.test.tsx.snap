// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render 1`] = `
<ForwardRef(Bootstrap(Form))
  onSubmit={[Function]}
>
  <ForwardRef(Bootstrap(FormGroup))
    controlId="formBasicUsername"
  >
    <ForwardRef(ContextTransform)>
      Username
    </ForwardRef(ContextTransform)>
    <ForwardRef(ContextTransform)
      defaultValue="a@b.com"
      onChange={[Function]}
      placeholder="Enter username"
      type="text"
    />
  </ForwardRef(Bootstrap(FormGroup))>
  <ForwardRef(Bootstrap(FormGroup))
    controlId="formBasicPassword"
  >
    <ForwardRef(ContextTransform)>
      Password
    </ForwardRef(ContextTransform)>
    <ForwardRef(ContextTransform)
      defaultValue="Password123"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
  </ForwardRef(Bootstrap(FormGroup))>
  <ForwardRef(Bootstrap(Button))
    data-test-id="submitBtn"
    disabled={false}
    type="submit"
    variant="primary"
  >
    Submit
  </ForwardRef(Bootstrap(Button))>
</ForwardRef(Bootstrap(Form))>
`;
